extensions/resource-deployment/src/localizedConstants.ts

```diff
@@ -37,5 +38,29 @@ export const optionsTypeRadioOrDropdown = localize('optionsTypeRadioOrDropdown',
 export const azdataEulaNotAccepted = localize('azdataEulaNotAccepted', "Deployment cannot continue. Azure Data CLI license terms have not yet been accepted. Please accept the EULA to enable the features that requires Azure Data CLI.");
 export const azdataEulaDeclined = localize('azdataEulaDeclined', "Deployment cannot continue. Azure Data CLI license terms were declined.You can either Accept EULA to continue or Cancel this operation");
 export const acceptEulaAndSelect = localize('deploymentDialog.RecheckEulaButton', "Accept EULA & Select");
+
+export const resourceTypePickerDialogTitle = localize('resourceTypePickerDialog.title', "Select the deployment options");
+export const resourceTypeSearchBoxDescription = localize('resourceTypePickerDialog.resourceSearchPlaceholder', "Filter resources...");
+export const resoucrceTypeCategoryListViewTitle = localize('resourceTypePickerDialog.tagsListViewTitle', 'Categories');
+
 export const scriptToNotebook = localize('ui.ScriptToNotebookButton', "Script");
 export const deployNotebook = localize('ui.DeployButton', "Run");
+
+export function getResourceTypeCategoryLocalizedString(resourceTypeCategory: string): string {
```

Comment:
Just make the ResourceTypeCategories enum values these localized strings. Then this function would not be needed at all as you can use the corresponding enum values directly.

Reply:
You can't set const enum values to computed values. Const enums are build-time only constructs and so computed strings such as the localize function aren't able to be used.

Reply:
(this is going with your other comment to make it a const enum).

And anyways - he still needs to have the constant non-localized values declared somewhere for use in the JSON. So you still need to have a lookup of some sort (unless you wanted to get real fancy and do annotations or whatever. But this is fine)